File: leetcode-26-Remove_Duplicates_from_Sorted_Array/leetcode-26-Remove_Duplicates_from_Sorted_Array.c
#include <stdio.h>
#include <stdlib.h>

int removeDuplicates(int* nums, int numsSize) {
    int num_of_dup = numsSize;

    for (int i = 0; i<num_of_dup; ++i){
        int dup = 0;
        for(int j = i+1; j<num_of_dup; ++j){
            if(nums[j] == nums[i] || dup){
                if(!dup){
                    dup = 1;
                }
                nums[j-1] = nums[j];
            }
        }
        if(dup){
            --i;
            --num_of_dup;
        }
    }
    return num_of_dup;
}

int main(){
    // [0,0,1,1,1,2,2,3,3,4]
    int nums[] = {0, 0, 1, 1, 1, 2, 2, 3, 3, 4};
    int numsSize = 10;

    int uniqe_values = removeDuplicates(nums, numsSize);

    printf("%d\n",uniqe_values);

    return EXIT_SUCCESS;
}

File: leetcode-26-Remove_Duplicates_from_Sorted_Array/leetcode-26-Remove_Duplicates_from_Sorted_Array.py
class Solution:
    def removeDuplicates(self, nums: list[int]) -> int:
        last_i: int = 0
        i: int = 0
        while i < len(nums):
            while i < len(nums) and nums[last_i] == nums[i]:
                i += 1
            if i < len(nums):
                last_i += 1
                nums[last_i], nums[i] = nums[i], nums[last_i]
                i += 1
        return last_i+1


if __name__ == "__main__":
    nums: list[int] = [1,1,2]
    result = Solution().removeDuplicates(nums)
    print(result)
